### PR TITLE
[improvement](agg) speed up serialization of aggregation function collect_list/collect_set

### PR DESCRIPTION
**Pull request by mrhhsg**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by mrhhsg** [comment]
run buildall

**Comment by mrhhsg** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]
LGTM

File: be/src/agent/be_exec_version_manager.h [hunk 1]
```diff
@@ -62,8 +62,10 @@ class BeExecVersionManager {
  *    a. aggregation function do not serialize bitmap to string.
  *    b. array contains/position/countequal function return nullable in less situations.
  *    c. cleared old version of Version 2.
+ * 4: start from doris 2.2
+ *    a. aggregation: collect_list does not serialize the aggregation data to string.
 */
-inline const int BeExecVersionManager::max_be_exec_version = 3;
+inline const int BeExecVersionManager::max_be_exec_version = 4;
```
Comment (JackDrogon) [inline be/src/agent/be_exec_version_manager.h:68]: why not use constexpr

File: be/src/vec/aggregate_functions/aggregate_function_collect.h [hunk 2]
```diff
@@ -535,6 +535,18 @@ class AggregateFunctionCollect
         if constexpr (ShowNull::value) {
             this->data(place).insert_result_into(to);
         } else {
+            if (version >= 4) {
```
Comment (JackDrogon) [inline be/src/vec/aggregate_functions/aggregate_function_collect.h:538]: what's the meaning of 4

File: be/src/vec/aggregate_functions/aggregate_function_collect.h [hunk 3]
```diff
@@ -547,6 +559,25 @@ class AggregateFunctionCollect
                 this->data(place).deserialize_and_merge(column, i);
             }
         } else {
+            if (version >= 4) {
```
Comment (JackDrogon) [inline be/src/vec/aggregate_functions/aggregate_function_collect.h:562]: too many special judge version >= 4, use factory method to refactor it

**Review by JackDrogon — CHANGES_REQUESTED** [review]

**Pull request closed without merging**